### PR TITLE
Add a new cmake option OLP_SDK_USE_LIBCRYPTO

### PR DESCRIPTION
**Pull request by mykhailo-kuchma**
A new option OLP_SDK_USE_LIBCRYPTO is used to conditionally enable or
disable the libcrypto dependency when needed.

Resolves: OLPEDGE-2735

Signed-off-by: Mykhailo Kuchma <ext-mykhailo.kuchma@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/CMakeLists.txt [hunk 1]
```diff
@@ -51,9 +51,10 @@ else()
     find_package(CURL REQUIRED)
 endif()
 
-
 set(Network_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
 
+option(OLP_SDK_USE_LIBCRYPTO "Enables the libcrypto dependency" ON)
```
Comment (andescu) [inline olp-cpp-sdk-core/CMakeLists.txt:56]: Should it not be in curl.cmake instead. It does not seem relevant to all other cmake includes.

**Review by andescu — APPROVED** [review]

**Review by HalynaDumych — APPROVED** [review]
